### PR TITLE
Add level back to tinylicious

### PR DESCRIPTION
**Pull request by kurtb**
Being able to store ops on disk, rather than fully in memory, makes tinylicious scale better as a service. The functionality used to exist but was removed due to an NPM package security warning. A prior version of level-sublevel does not take a dependency on the package with the security warning.

### CODE REVIEW

**Review by tanviraumi — APPROVED** [review]
Looks good! Thanks for bringing it back 👍

File: server/tinylicious/src/services/dbFactory.ts [hunk 1]
```diff
@@ -4,10 +4,16 @@
...
+import { Provider } from "nconf";
 import { InMemoryDb } from "./inMemorydb";
+import { LevelDb } from "./levelDb";
 
 export class DbFactory implements IDbFactory {
-    private readonly db = new InMemoryDb();
+    private readonly db;
+
+    constructor(config: Provider) {
+        this.db = config.get("db:inMemory") ? new InMemoryDb() : new LevelDb(config.get("db:path"));
```
Comment (anthony-murphy) [inline server/tinylicious/src/services/dbFactory.ts:15]: What will the default be? For back compat reasons with our existing usages the default should be inMemory

Comment (tanviraumi) [reply server/tinylicious/src/services/dbFactory.ts:15]: Good point! Even though the config says inMemory: true, we will want to flip it have levelDb as an option. In every other case, use the inMemoryDB.

@kurtb: Can you do it?

Comment (kurtb) [reply server/tinylicious/src/services/dbFactory.ts:15]: Default config is inMemory

**Review by SamBroner — APPROVED** [review]
Looks good. @TylerButler had feedback with regards to documentation and testing. I'll let him follow up.

**Comment by tylerbutler** [comment]
@kurtb Thanks for the contribution! We're working to ensure all code introduced to the repo has documentation and some tests to help protect it from breaking. This seems especially important in this case since this code isn't exercised in any of our core scenarios today. Could you add some tests and update the Tinylicious readme with information about how to use this?

**Comment by markfields** [comment]
Also I'd love to see the corresponding `@types/` packages pulled in so we can have less `any`.   https://www.typescriptlang.org/dt/search?search=types%2Flevel

**Comment by kurtb** [comment]
This is simply a bug fix and not the introduction of new functionality.

This PR is a revert of #3463 which was made due to "a moderate vulnerability in npm audit, leading to an unpleasant warning for new devs trying out our hello world flow." The only actual change made is using "level-sublevel": "6.6.4" rather than the previous version. Everything else isn't new code.

level-sublevel itself does not have the npm audit warning but a package it depends on. The earlier version does not take a dependency on this and it's not needed for tinylicious.

**Comment by tylerbutler** [comment]
@kurtb Thanks for the clarification. Given the history of this code and the state of Tinylicious testing at the moment, testing isn't needed. I would like to update the README with some info about how to activate this. Or is it on by default? What config changes do I need to make? Do I need to specify the path for the db, for example?

I'll merge this in in a few hours. Feel free to push new commits here, or follow up with another PR readme update if that's easier.

Thanks for being patient with us! We're doing a lot of new things at the moment and I'm sorry for the lack of communication on this PR.

**Comment by kurtb** [comment]
Thanks @tylerbutler! I also noticed @markfields added some starter tests - thanks! Once you merge this I'll follow up in the next day or so with the README updates and hopefully some tests too.

**Pull request merged**